[SEC-31445] severity modifiers doc (#37170)
DataDog/documentation: content/en/security/automation_pipelines/modify_severity.md

---
title: Severity Modifier Rules
products:
  - name: Cloud Security
    url: /security/cloud_security_management/
    icon: cloud-security-management
  - name: Code Security
    url: /security/code_security/
    icon: security-code-security
  - name: App and API Protection
    url: /security/application_security/
    icon: app-sec
further_reading:
  - link: "/security/automation_pipelines"
    tag: "Documentation"
    text: "Automation Pipelines"
---

{{< product-availability >}}

Configure severity modifier rules to adjust finding severities to reflect your organization's business context. For example, downgrade findings on isolated environments to reduce noise, or upgrade findings on databases containing personally identifiable information (PII) so they receive immediate attention.

## Create a severity modifier rule

> 1. In Datadog, go to **Security** > **Settings** > [Findings Automation][2]. Click **Add a New Rule**, then select **Modify Severity Level**. The Create a New Rule page opens.

Reviewer: ```suggestion
1. In Datadog, go to **Security** > **Settings** > [**Findings Automation**][2]. Click **Add a New Rule**, then select **Modify Severity Level**. The Create a New Rule page opens.
```

1. Under **Rule name**, enter a descriptive name for the rule; for example, "Increase severity for services accessing PII databases".
1. Add your rule criteria into the following fields:
    - **Any of these types**: The types of findings that the rule should check for. Available types include:
      - Runtime Code Vulnerability
      - Static Code Vulnerability
      - Library Vulnerability
      - Secret
      - Infrastructure as Code
      - Container Image Vulnerability
      - Host Vulnerability
      - Misconfiguration
      - Attack Path
      - Identity Risk
      - API Security
    - **Any of these tags or attributes**: The resource tags or attributes that must match for the rule to apply.

> 1. To add severity criteria to the rule, click **Add Severity**.

Reviewer 2: nit: this is the sentence we have for other automation rules, but in the context severity modifiers, the notion of "severity criteria" it could be a bit confusing. May be worth rephrasing it more clearly.

1. Define the severity modification action:

>     - **Set to a specific level**: Sets matching findings to a fixed severity. Choose from **Info / None**, **Low**, **Medium**, **High**, or **Critical**. **Info / None** is only valid for finding types whose lowest severity is **Info** or **None**. Using it in a rule that targets finding types with **Low** as their lowest severity (Infrastructure as Code, Runtime Code Vulnerability, Secret, Static Code Vulnerability) results in a validation error. See [Severity floors by finding type](#severity-floors-by-finding-type).

Reviewer 2: > Using it in a rule that targets finding types with **Low** as their lowest severity (Infrastructure as Code, Runtime Code Vulnerability, Secret, Static Code Vulnerability) results in a validation error.

This information is duplicated in the section `#severity-floors-by-finding-type`. IMO this is very specific information about an edge case, I'm not sure it's worth talking about it directly in this high level configuration overview. I'd say that just redirecting to `#severity-floors-by-finding-type` for more detail is the right move.

1. Optionally, enter a **Description** explaining why the rule applies. This text appears in the severity breakdown panel when a user views a modified finding.
1. Click **Save**. The rule applies to new findings immediately and starts checking existing findings within the next hour.

**Note**: You cannot use `@severity` or `@severity_details.user_adjusted` in the rule query. Severity modifier rules are evaluated against the Datadog-adjusted severity (`@severity_details.adjusted.value`), not the `@severity` value stored in the finding.

## Evaluation order

Severity modifier rules are the first step in the automation pipeline and run before mute, due date, inbox, and ticket creation rules. Within severity modifier rules, Datadog uses a first-match policy: findings are evaluated against your rules in order, and the first matching rule is applied. No further severity modifier rules are evaluated for that finding.

Because severity modifier rules run first, all downstream automation rules—including mute rules—see the modified severity when they are evaluated.

> ## Identify modified findings

Reviewer 2: this could be worth adding relevant example screenshot(s)?

Author: I added two TODO's, I'll add the screenshots once the frontend is ready.

- **Set action**: If the **Unknown** severity is included in the rule's severity selector, the rule matches and replaces **Unknown** with the specified target severity. You cannot set a finding's severity to **Unknown** using a severity modifier rule.

## Vulnerability findings and CVSS scores

For vulnerability findings that have a Datadog-adjusted CVSS score, a severity modifier also updates the adjusted score stored in `@severity_details.user_adjusted`. The updated score is set to approximately the midpoint of the target severity's CVSS v3 range:

| Target severity | CVSS v3 range |
|---|---|
| None | 0.0 |
| Low | 0.1–3.9 |
| Medium | 4.0–6.9 |
| High | 7.0–8.9 |
| Critical | 9.0–10.0 |

The original CVSS vector is never modified. No synthetic vector is generated to match the adjusted score.

## Auto-closed and passed findings

Severity modifiers are not cleared or updated for findings that transition to auto-closed or where the evaluation result is **pass**. If the rule that originally modified a finding is later edited or deleted, those findings retain the severity that was set when they closed.

> If you view an auto-closed or passed finding whose severity modifier rule was subsequently deleted, the side panel may reference a rule that no longer exists.

Reviewer 2: not sure if we usually document/acknowledge those kinds of edge cases shortcomings

## Create a severity modifier rule

1. In Datadog, go to **Security** > **Settings** > [**Findings Automation**][2]. Click **Add a New Rule**, then select **Modify Severity Level**. The Create a New Rule page opens.
1. Under **Rule name**, enter a descriptive name for the rule; for example, "Increase severity for services accessing PII databases".
1. Add your rule criteria into the following fields:
    - **Any of these types**: The types of findings that the rule should check for. Available types include:
      - Runtime Code Vulnerability
      - Static Code Vulnerability
      - Library Vulnerability
      - Secret
      - Infrastructure as Code
      - Container Image Vulnerability
      - Host Vulnerability
      - Misconfiguration
      - Attack Path
      - Identity Risk
      - API Security
      - Workload Activity
    - **Any of these tags or attributes**: The resource tags or attributes that must match for the rule to apply.
1. Optionally, click **Add Severity** to filter findings by severity level. The rule matches against each finding's Datadog-adjusted severity, before any user-defined adjustments.
1. Define the severity modification action:

>     - **Set to a specific level**: Sets matching findings to a fixed severity. Choose from **Info / None**, **Low**, **Medium**, **High**, or **Critical**. **Info / None** is not valid for all finding types; see [Severity floors by finding type](#severity-floors-by-finding-type).

Reviewer: ```suggestion
    - **Set to a specific level**: Sets matching findings to a fixed severity. Choose from **Info / None**, **Low**, **Medium**, **High**, or **Critical**.
      <div class="alert alert-info"><strong>Info / None</strong> is only valid for some finding types; see <a href="#severity-floors-by-finding-type">Severity floors by finding type</a>.</div>
```
I thought this would be easier to parse with stating it in the positive ("only valid for some" rather than "not valid for all") and by putting it into a separate not callout:
<img width="739" height="154" alt="Image" src="https://github.com/user-attachments/assets/2b666aff-3d24-431c-a2a0-3eec01d0b22a" />

1. Optionally, enter a **Description** explaining why the rule applies. This text appears in the severity breakdown panel when a user views a modified finding.
1. Click **Save**. The rule applies to new findings immediately and starts checking existing findings within the next hour.

**Note**: You cannot use `@severity` or `@severity_details.user_adjusted` in the rule query. Severity modifier rules are evaluated against the Datadog-adjusted severity (`@severity_details.adjusted.value`), not the `@severity` value stored in the finding.

## Evaluation order

Severity modifier rules are the first step in the automation pipeline and run before mute, due date, inbox, and ticket creation rules. Within severity modifier rules, Datadog uses a first-match policy: findings are evaluated against your rules in order, and the first matching rule is applied. No further severity modifier rules are evaluated for that finding.

> A rule counts as a match only if applying its action would change the finding's severity. If the action would leave the severity unchanged—for example, a shift action that has already reached a severity bound, or a set action that targets the finding's current severity—the rule does not match, and Datadog continues evaluating subsequent severity modifier rules for that finding.

Reviewer: ```suggestion
A rule counts as a match only if applying its action would change the finding's severity. If the action would leave the severity unchanged (for example, a shift action that has already reached a severity bound, or a set action that targets the finding's current severity), the rule does not match, and Datadog continues evaluating subsequent severity modifier rules for that finding.
```

Because severity modifier rules run first, all downstream automation rules—including mute rules—see the modified severity when they are evaluated.

## Identify modified findings

> **TODO:** Add a screenshot of a severity pill pop-over in an explorer.

Reviewer: ```suggestion
<!-- **TODO:** Add a screenshot of a severity pill pop-over in an explorer. -->
```
Totally okay to have these in here, but for due diligence's sake, I should hold off on approving until they're at least commented out 🙂

Findings affected by a severity modifier rule display a visual indicator in explorer list views and in the finding's side panel header. Hovering over the indicator shows the automation rule responsible for the change.

> **TODO:** Add a screenshot of a severity breakdown in side-panel.

Reviewer: ```suggestion
<!-- **TODO:** Add a screenshot of a severity breakdown in side-panel. -->
```

A rule counts as a match only if applying its action would change the finding's severity. If the action would leave the severity unchanged (for example, a shift action that has already reached a severity bound, or a set action that targets the finding's current severity), the rule does not match, and Datadog continues evaluating subsequent severity modifier rules for that finding.

Because severity modifier rules run first, all downstream automation rules—including mute rules—see the modified severity when they are evaluated.

## Identify modified findings

{{< img src="security/automation_pipelines/severity_pill_popover.png" alt="A findings explorer list with a severity pill that has a modifier indicator, showing a pop-over that names the automation rule responsible for adjusting the finding's severity" style="width:100%;" >}}

> Findings affected by a severity modifier rule display a visual indicator in explorer list views and in the finding's side panel header. Hovering over the indicator shows the automation rule responsible for the change.

Reviewer: ```suggestion
Findings affected by a severity modifier rule display a visual indicator in explorer list views and in the finding's side panel header. Hovering over the indicator shows the automation rule responsible for the change:

{{< img src="security/automation_pipelines/severity_pill_popover.png" alt="A findings explorer item that displays a severity pill with a modifier indicator. A pop-over provides more information about the automation rule responsible for adjusting the finding's severity" style="width:65%;" >}}
```
I think it makes it easier to understand an image if some text provides some context to it beforehand. I also tweaked the alt text to make it a little easier to parse, and shrunk the image down a bit so it would look a little more proportional with the rest of the page content 🙂

{{< img src="security/automation_pipelines/severity_breakdown.png" alt="A finding side panel showing the severity breakdown, with the original severity, CVSS score, and CVSS vector, the automation rule that triggered the change, and the resulting severity level and adjusted CVSS score" style="width:100%;" >}}

For findings that have a CVSS score (Container Image Vulnerability, Host Vulnerability, Library Vulnerability, and Runtime Code Vulnerability), the side panel severity section also includes a breakdown showing:
- The original severity level, CVSS score, and CVSS vector before modification.
- The name of the automation rule that triggered, with a direct link to the rule.

> - The resulting severity level and adjusted CVSS score.

Reviewer: ```suggestion
For findings that have a CVSS score (Container Image Vulnerability, Host Vulnerability, Library Vulnerability, and Runtime Code Vulnerability), the side panel severity section also includes a breakdown showing:
- The original severity level, CVSS score, and CVSS vector before modification.
- The name of the automation rule that triggered, with a direct link to the rule.
- The resulting severity level and adjusted CVSS score.

{{< img src="security/automation_pipelines/severity_breakdown.png" alt="A finding side panel showing the severity breakdown, with the original severity, CVSS score, and CVSS vector; the automation rule that triggered the change; and the resulting severity level and adjusted CVSS score" style="width:100%;" >}}
```
Another small edit to reverse the edit and to make some really picky punctuation edits
